### PR TITLE
 messages update for passwords management policy

### PR DESCRIPTION
**Pull request by marwanehcine**
Update cas to make password policy management option available for georchestra. change password URL can be set in default.properties file.

### CODE REVIEW

File: src/main/resources/messages.properties [hunk 1]
```diff
@@ -287,16 +287,16 @@ screen.surrogates.button.cancel=Cancel
...
 screen.accountdisabled.heading=This account has been disabled.
 screen.accountdisabled.message=Please contact the system administrator to regain access.
 screen.accountlocked.heading=This account has been locked.
 screen.accountlocked.message=Please contact the system administrator to regain access.
 screen.expiredpass.heading=Your password has expired.
-screen.expiredpass.message=Please <a href="{0}">change your password</a>.
+screen.expiredpass.message=Please <a href="https://georchestra-127-0-1-1.traefik.me/console/account/passwordRecovery">change your password</a>.
 screen.mustchangepass.heading=You must change your password.
-screen.mustchangepass.message=Please <a href="{0}">change your password</a>.
+screen.mustchangepass.message=Please <a "href=https://georchestra-127-0-1-1.traefik.me/console/account/passwordRecovery">change your password</a>.
```
Comment (pmauduit) [inline src/main/resources/messages.properties:299]: The URLs will only be valid in the context of the docker composition. How about using host-relative urls instead ?
("https://..." -> "/console/...").

Comment (pmauduit) [reply src/main/resources/messages.properties:299]: Also if it was templated (`{0}`) before your modification on line 297 & 299, maybe CAS already has a way to provide the expected url ?

**Review by pmauduit — CHANGES_REQUESTED** [review]

File: src/main/resources/messages.properties (hunk 1 above)
Comment (marwanehcine) [reply src/main/resources/messages.properties:299]: @pmauduit , In fact Urls was defined inside dependency cas-server-support-thymeleaf-6.3.7.4.jar which is extracted at gradle build process to generate cas templates

Comment (marwanehcine) [reply src/main/resources/messages.properties:299]: > The URLs will only be valid in the context of the docker composition. How about using host-relative urls instead ? ("https://..." -> "/console/...").

I think URL should be configurable, at build process or something like that

File: Dockerfile [hunk 2]
```diff
@@ -9,8 +9,8 @@ VOLUME [ "/tmp", "/run/jetty" ]
 
 EXPOSE 8080
 
-COPY build/cas /var/lib/jetty/webapps/cas
-
+COPY --chmod=777 build/cas /var/lib/jetty/webapps/cas
```
Comment (pmauduit) [inline Dockerfile:12]: why such wide permissions ?

File: src/main/java/org/georchestra/cas/MessagesConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+class MessagesConfiguration implements CommandLineRunner {
+
+    @Autowired GeorchestraConfiguration georchestraConfiguration;
+    public static final String FILE_PATH = "webapps/cas/WEB-INF/classes/messages.properties";
+    public static final String EXPIRE_PASS_KEY = "screen.expiredpass.message";
+    public static final String EXPIRE_PASS_VALUE = "Please <a href=\"%s\">change your password</a>.";
+    public static final String WARNING_PASS_KEY = "password.expiration.warning";
+    public static final String WARNING_PASS_VALUE = "Your password expires in {0} day(s). Please <a href=\"%s\">change your password</a> now.";
+    public static final String MUST_CHANGE_PASS_KEY = "screen.mustchangepass.message";
+    public static final String MUST_CHANGE_PASS_VALUE = "Please <a href=\"%s\">change your password</a>.";
```
Comment (pmauduit) [inline src/main/java/org/georchestra/cas/MessagesConfiguration.java:26]: the mix between templating syntax (format string `%s` vs `{0}`) looks weird to me

File: src/main/java/org/georchestra/cas/MessagesConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+
+@Component
+@Order(1)
+class MessagesConfiguration implements CommandLineRunner {
```
Comment (pmauduit) [inline src/main/java/org/georchestra/cas/MessagesConfiguration.java:17]: Why do you need to implement `CommandLineRunner` ?

**Review by pmauduit — COMMENTED** [review]
I can't approve nor refuse the PR, because I definitely lack context of what we want to achieve here, It'd take me some time to get into it.

File: src/main/java/org/georchestra/cas/MessagesConfiguration.java (hunk 4 above)
Comment (marwanehcine) [reply src/main/java/org/georchestra/cas/MessagesConfiguration.java:17]: I need it to start updating message properties when application has already started

File: src/main/java/org/georchestra/cas/MessagesConfiguration.java (hunk 3 above)
Comment (marwanehcine) [reply src/main/java/org/georchestra/cas/MessagesConfiguration.java:26]: {0} is related to messages.property (defined there) and %s is used here to substitute goal

File: Dockerfile (hunk 2 above)
Comment (marwanehcine) [reply Dockerfile:12]: I will check if 700 is enough

Comment (marwanehcine) [reply Dockerfile:12]: @pmauduit  I have checked all possibility and app works only for 777, so I kept it like this. Thanks

**Pull request merged**